server/src/main/java/org/apache/cassandra/sidecar/tasks/CdcRawDirectorySpaceCleaner.java

```diff
@@ -182,10 +182,10 @@ protected void routineCleanUp()
     {
         for (InstanceMetadata instanceMetadata : instancesMetadata.instances())
         {
-            List<String> dataDirectories = instanceMetadata.dataDirs();
-            dataDirectories.stream()
-                           .map(dir -> new File(dir, CDC_DIR_NAME))
-                           .forEach(this::cleanUpCdcRawDirectory);
+            if (instanceMetadata.cdcDir() != null)
```

Comment:
how about add a warning log message if cdcDir is null?
`CdcRawDirectorySpaceCleaner` is enabled explicitly via a property; if it is enabled, but cdc dir does not exist, it likely indicates some issue.

Reply:
can we avoid using Optional here? I have the following suggestion

```
            String cdcDir = instanceMetadata.cdcDir();
            if (cdcDir != null)
            {
                try
                {
                    cleanUpCdcRawDirectory(new File(cdcDir));
                }
                catch (Exception e)
                {
                    LOGGER.warn("Unable to clean up CDC directory {}", cdcDir, e);
                }
            }
            else
            {
                LOGGER.warn("CDC directory is not configured. Skipping clean up");
            }
        }
```